### PR TITLE
Remove  --host-whitelist 2.0

### PR DESCRIPTION
**Pull request by JanetMo**
remove the --host-whitelist option, issue #7761

deleted the command and it's associated tests, the changelog was already adapted

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 1]
```diff
@@ -592,14 +592,6 @@ static class PrivacyOptionGroup {
       defaultValue = "localhost,127.0.0.1")
   private final JsonRPCAllowlistHostsProperty hostsAllowlist = new JsonRPCAllowlistHostsProperty();
 
-  @Option(
-      names = {"--host-whitelist"},
-      hidden = true,
-      paramLabel = "<hostname>[,<hostname>...]... or * or all",
-      description =
-          "Deprecated in favor of --host-allowlist. Comma separated list of hostnames to allow for RPC access, or * to accept any host (default: ${DEFAULT-VALUE})")
-  private final JsonRPCAllowlistHostsProperty hostsWhitelist = new JsonRPCAllowlistHostsProperty();
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:601]: There is also some code in the configure method of BesuCommand that uses the hostsWhitelist variable that needs to be removed as well

Comment (JanetMo) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:619]: Thanks for the review! You mean this part further down in the same file, of another one?

```
    // hostsWhitelist is a hidden option. If it is specified, add the list to hostAllowlist
    if (!hostsWhitelist.isEmpty()) {
      // if allowlist == default values, remove the default values
      if (hostsAllowlist.size() == 2
          && hostsAllowlist.containsAll(List.of("localhost", "127.0.0.1"))) {
        hostsAllowlist.removeAll(List.of("localhost", "127.0.0.1"));
      }
      hostsAllowlist.addAll(hostsWhitelist);
    }
```

File: benchmark/docker-compose.yml [hunk 2]
```diff
@@ -66,7 +66,6 @@ services:
     - --rpc-ws-enabled
     - --rpc-ws-apis=admin,eth,miner,web3,net,priv,eea
     - --rpc-ws-host=0.0.0.0
-    - --host-whitelist=*
```
Comment (macfarla) [inline benchmark/docker-compose.yml:69]: instead of deleting this line, can you change to host-allowlist

File: benchmark/docker-compose.yml [hunk 3]
```diff
@@ -84,7 +83,6 @@ services:
...
```
Comment (macfarla) [inline benchmark/docker-compose.yml:87]: instead of deleting this line, can you change to host-allowlist

**Review by macfarla — COMMENTED** [review]
needs a changelog entry since this is a breaking change

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 1 above)
Comment (JanetMo) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:619]: @macfarla Thanks, unfortunately we were still not able to run the repo locally... Could you guide us on which piece of code to remove?

**Comment by JanetMo** [comment]
@macfarla I think the changelog entry was already done in this PR https://github.com/hyperledger/besu/pull/7762

File: benchmark/docker-compose.yml (hunk 2 above)
Comment (JanetMo) [reply benchmark/docker-compose.yml:69]: Alright, I changed it to host-allowlist

File: benchmark/docker-compose.yml (hunk 3 above)
Comment (JanetMo) [reply benchmark/docker-compose.yml:87]: Changed ad well, thanks for reviewing!

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 1 above)
Comment (macfarla) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:619]: @JanetMo feel free to post in [discord](https://discord.com/channels/905194001349627914/938504958909747250) if you're having trouble compiling or building locally, plenty of folks there that can help you out

Comment (JanetMo) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:619]: @macfarla Thanks, appreciate the support. I retried but still wasn't able to build it locally, is there a way to unassign from the issue? I think I'm not up for it, and probably holding you back on with this one.

Comment (macfarla) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:619]: ok yes I can unassign you @JanetMo - you're welcome back any time - but being able to compile the code is a must :)

**Comment by macfarla** [comment]
closing this PR at request of @JanetMo and it's now assigned to @pr9t

**Pull request closed without merging**